### ISSUE TITLE
Schema update unnecessarily drops/creates related objects when Sensitivity Classification changes are detected

### ISSUE THREAD

**Issue by tmenier** [microsoft/DacFx#48]
**Summary:**

When schema compare (correctly) detects that a [Sensitivity Classification](https://docs.microsoft.com/en-us/sql/relational-databases/security/sql-data-discovery-and-classification) should be created, dropped, or modified in the target, the generated script takes the unnecessary steps of dropping/recreating all indexes and constraints related to that column. Index rebuilding in particular is problematic in our ability to perform zero-downtime/zero-disruption production releases. Since sensitivity classifications currently [cannot be ignored](https://github.com/microsoft/DacFx/issues/46) by schema compare, it seems our only recourse is to ensure these classifications are always in sync between the dacpac and target databases, without relying on DacFx to help with this. This is a little painful given our security/ops people apply Azure-recommended classifications directly to the target Azure databases from time to time.

- DacFx Version: 150.5282.3
- .NET Framework (Windows-only) or .NET Core: .NET 5
- Environment (local platform and source/target platforms):
  - Windows 10
  - Visual Studio 2019 (v16.11.5)
  - SQL project with DSP `Microsoft.Data.Tools.Schema.Sql.Sql150DatabaseSchemaProvider` or `Microsoft.Data.Tools.Schema.Sql.SqlAzureV12DatabaseSchemaProvider`
  - .NET 5 console app with Microsoft.SqlServer.DacFx v150.5282.3 installed
  - Target database SQL Server 2019 or Azure SQL

**Steps to Reproduce:**

(easiest if you already have a SQL project and sync'd live database to experiment with)

1. Define a table in a SQL project that contains an index and constraint on some column, build dacpac.
2. Create a target database (SQL 2019 or Azure SQL) with the same table defined (manually or by deploying the dacpac).
3. In the target database, add a sensitivity classification ([with T-SQL](https://docs.microsoft.com/en-us/sql/t-sql/statements/add-sensitivity-classification-transact-sql) or otherwise) to a column that has an index and constraint.
4. Generate a schema update script with the following code (or similar):
```c#
var source = new SchemaCompareDacpacEndpoint(dacpacPath);
var target = new SchemaCompareDatabaseEndpoint(targetConnectionString);
var comp = new SchemaComparison(source, target);
var script = comp.Compare().GenerateScript(dbName).Script;
```
5. Inspect generated script. Notice it correctly drops the classification, but it also unnecessarily drops and recreates any indexes and constraints defined on that column.

**Did this occur in prior versions?  If not - which version(s) did it work in?**

Yes, same behavior in older versions of DacFx published to NuGet. I'm fairly certain this bug has existed since support for sensitivity classifications was first introduced.

**Comment by zijchen** [comment]
@tmenier We introduced `IgnoreSensitivityClassifications` DacDeployOption in [Microsoft.SqlServer.DacFx 160.6161.0 ](https://www.nuget.org/packages/Microsoft.SqlServer.DacFx/160.6161.0), you can specify it during Schema Compare too.

```cs
var comp = new SchemaComparison(source, target);
comp.Options.IgnoreSensitivityClassifications = true;
var script = comp.Compare().GenerateScript(dbName).Script;
```

**Comment by llali** [comment]
@zijchen  is this shipped in 19.1? can we close it?
